src/implot_visualization/app/main.cpp

```diff
@@ -122,9 +128,20 @@ static void main_loop(void *arg) {
     ImGui_ImplSDL2_NewFrame();
     ImGui::NewFrame();
 
-    // Visualize One GR SignalBuffer
+    // FAIR Visualization
     if (visualize_gr_signal) {
-        fetch("http://localhost:8080/pulsed_power/Acquisition?channelNameFilter=sinus@4000Hz,saw@4000Hz,square@4000Hz", signals, &deserializer);
+        // Periodic polling
+        for (int i = 0; i < subscriptions.size(); i++) {
+            if (i == 0) {
+                if (subscriptions[subscriptions.size() - 1].fetchFinished) {
+                    subscriptions[i].fetch();
+                }
+            } else {
+                if (subscriptions[i - 1].fetchFinished) {
+                    subscriptions[i].fetch();
+                }
+            }
+        }
```

Comment:
I think it should be possible to have multiple different subscriptions pending at the same time. This would simplify/decouple the code for the different subscriptions while allowing different update rates for them and prevent one unresponsive service worker from blocking the other subscriptions.

Reply:
Yes, you are right! Fixed it in a following commit.